packages/grid/_modules_/grid/hooks/features/clipboard/useGridClipboard.ts

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import { ownerDocument } from '@material-ui/core/utils';
+import { GridApiRef } from '../../../models/api/gridApiRef';
+import { useGridApiEventHandler } from '../../root/useGridApiEventHandler';
+import { GRID_KEYDOWN } from '../../../constants/eventsConstants';
+import { buildCSV } from '../export/serializers/csvSerializer';
+import { useGridSelector } from '../core/useGridSelector';
+import { visibleGridColumnsSelector } from '../columns/gridColumnsSelector';
+import { gridCheckboxSelectionColDef } from '../../../models/colDef';
+import { GridClipboardApi } from '../../../models/api';
+import { useGridApiMethod } from '../../root/useGridApiMethod';
+
+export const useGridClipboard = (apiRef: GridApiRef): void => {
+  const visibleColumns = useGridSelector(apiRef, visibleGridColumnsSelector);
+
+  const writeToClipboardPolyfill = React.useCallback(
+    (data: string) => {
+      const doc = ownerDocument(apiRef.current.rootElementRef!.current!);
+      const span = doc.createElement('span');
+      span.style.whiteSpace = 'pre';
+      span.style.userSelect = 'all';
+      span.style.opacity = '0px';
+      span.textContent = data;
+
+      apiRef.current.rootElementRef!.current!.appendChild(span);
+
+      const range = doc.createRange();
+      range.selectNode(span);
+      const selection = window.getSelection();
+      selection!.removeAllRanges();
+      selection!.addRange(range);
+
+      try {
+        doc.execCommand('copy');
+      } finally {
+        apiRef.current.rootElementRef!.current!.removeChild(span);
+      }
+    },
+    [apiRef],
+  );
+
+  const copySelectedRowsToClipboard = React.useCallback(
+    (includeHeaders = false) => {
+      const selectedRows = apiRef.current.getSelectedRows();
+      const filteredColumns = visibleColumns.filter(
+        (column) => column.field !== gridCheckboxSelectionColDef.field,
+      );
+
+      if (selectedRows.size === 0 || filteredColumns.length === 0) {
+        return;
+      }
+
+      const data = buildCSV({
+        columns: visibleColumns,
+        rows: selectedRows,
+        includeHeaders,
+        getCellParams: apiRef.current.getCellParams,
+        delimiterCharacter: '\t',
+      });
+
+      if (navigator.clipboard) {
+        navigator.clipboard.writeText(data).catch(() => {
+          writeToClipboardPolyfill(data);
+        });
+      } else {
+        writeToClipboardPolyfill(data);
+      }
+    },
+    [apiRef, visibleColumns, writeToClipboardPolyfill],
+  );
+
+  const handleKeydown = React.useCallback(
+    (event: KeyboardEvent) => {
+      const isModifierKeyPressed = event.ctrlKey || event.metaKey || event.altKey;
+      if (event.key.toLowerCase() !== 'c' || !isModifierKeyPressed) {
+        return;
+      }
+
+      const selection = window.getSelection();
+      const anchorNode = selection?.anchorNode;
+      if (anchorNode && apiRef.current.windowRef?.current?.contains(anchorNode)) {
```

Comment:
I don't think that it's working. As soon as I select and unselect something, ctrl + c do no longer work.

Reply:
If the focus is not inside the grid it won't work. Can you check if that's the case? We can't listen to the events in the window.

Reply:
I don't know, for me, I can almost, not use the feature

https://user-images.githubusercontent.com/3165635/123002156-e172c780-d3b1-11eb-81b3-3f54837042bf.mp4

Reply:
Fixed. There's always a `Selection` object, but I was not checking if the selected string is not empty.